packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -190,6 +200,34 @@ export const visualizeSharedString: VisualizeSharedObject = async (
 	};
 };
 
+/**
+ * {@link VisualizeSharedObject} for {@link SharedTree}.
+ */
+export const visualizeSharedTree: VisualizeSharedObject = async (
+	sharedObject: ISharedObject,
+	visualizeChildData: VisualizeChildData,
+): Promise<FluidObjectTreeNode> => {
+	const sharedTree = sharedObject as ISharedTree;
+
+	console.log("Default Visualizer :", JSON.stringify(sharedTree.root));
+
+	// TODO: Find Shared Tree API for JSON data format.
+	const children: Record<string, VisualChildNode> = {};
+	if (sharedTree.root !== undefined) {
+		for (const [key, value] of Object.entries(sharedTree.root)) {
```

Comment:
Is there a more type-safe API we can use to walk the tree's children? Or is this the canonical way to walk a SharedTree's contents (when you don't have a schema)?

Reply:
Assuming you want a simple readonly view there are a couple of approaches.

1. If you want the verbose view where value fields (always exactly one child) arn't inlined and such, tree.context.root to get the root field. Use .length and getNode(index) to read the nodes. Iterate the node to get the fields from it, and use the [valueSymbol]  and [typeNameSymbol] to get the rest of the data from the node.
2. If you want instead the more concise version with primitive nodes replaces with just primitive types and value fields inclined and such, tree.root to get the root. Use isEditiableField to tell if you got a field or node. If field, iterate it to get its children. If node get the field gets with object.onwKeys I think it is, then read those fields off the object with getField symbol.

Reply:
Also note that technically you always have a schema (one is stored in the document, you can read it via the storedSchema field. What you don't have is a statically known schema.

Reply:
`DefaultVisualizer` now uses `iterateNodes` and `iterateFields` for iterating through the tree-data.